packages/schema-extrct/src/json-schema-types.ts

```diff
@@ -72,36 +74,21 @@ export type ModuleSchema<T extends  SchemaTypes = SchemaTypes> = Schema<T> & {
 }
 
 export type FunctionSchema = Schema & {
-    $ref:typeof FunctionSchemaId;
+    $ref:typeof FunctionSchemaId | typeof ClassConstructorSchemaId;
     arguments:Schema[],
     restArgument?:Schema<'array'>,
-    returns:Schema
-}
-
-export type ClassConstructorSchema = Schema & {
-    $ref:typeof ClassConstructorSchemaId;
-    arguments:Schema[];
-    restArgument?:Schema<'array'>;
-    returns:Schema;
-    properties:{[name:string]:Schema};
-    extends?:Schema
+    returns?:Schema
 }
 
 export type ClassSchema = Schema & {
     $ref:typeof ClassSchemaId;
-    constructor:Schema;
+    constructor?:FunctionSchema;
```

Comment:
We should probably decide if constructor should be under FunctionSchema or something else.